File: week02/스택/10828.py
# 5가지 명령에 따라 출력
import sys
input = sys.stdin.readline

# 명령의 수
N = int(input())

stack = []

def push(arr, X):
  arr.append(X)

def pop(arr):
  if len(arr) == 0:
    print(-1)
  else:
    print(arr[-1])
    arr.pop()
    

def size(arr):
  print(len(arr))

def empty(arr):
  if len(arr) == 0:
    print(1)
  else:
    print(0)

def top(arr):
  if len(arr) == 0:
    print(-1)
  else:
    print(arr[-1])


for i in range(N):
  inst = input().split()
  # 명령어를 받을때 어떻게 입맛에 맞게 바꿀까
  if inst[0] == 'push':
    push(stack, int(inst[1]))
  elif inst[0] == 'top':
    top(stack)
  elif inst[0] == 'size':
    size(stack)
  elif inst[0] == 'pop':
    pop(stack)
  elif inst[0] == 'empty':
    empty(stack)

File: week02_JS/스택/10828.js
// JS로 스택 구현
const fs = require('fs');
const input = fs.readFileSync('input.txt').toString().trim().split('\n');

let N = parseInt(input[0])
let stack = []
let result = []


for (let i = 1; i <= N; i++) {
  let [order, num] = input[i].split(' ')
  if (order == 'push') {
    push(parseInt(num))
  } else if (order == 'pop') {
    pop()
  } else if (order == 'size') {
    size()
  } else if (order == 'empty') {
    empty()
  } else if (order == 'top') {
    top()
  }
}


function push(N) {
  stack.push(N)
}

function pop() {
  if (stack.length == 0) {
    result.push(-1)
  } else {
    result.push(stack.pop())
  }
}

function size() {
  result.push(stack.length)
}

function empty() {
  if (stack.length == 0) {
    result.push(1)
  } else {
    result.push(0)
  }
}

function top() {
  if (stack.length == 0) {
    result.push(-1)
  } else {
    result.push(stack[stack.length - 1])
  }
}

console.log(result.join('\n'))